tmt/base.py

```diff
@@ -2908,6 +2908,28 @@ def to_serialized(self) -> Dict[str, Any]:
     def from_serialized(cls, serialized: Dict[str, Any]) -> 'Result':
         return cls(serialized, name=serialized.pop('name'))
 
+    @staticmethod
+    def failures(log: Optional[str], msg_type: str = 'FAIL') -> str:
+        """ Filter stdout and get only messages with certain type """
+        if not log:
+            return ''
+        filtered = ''
+
+        # Filter beakerlib style logs
```

Comment:
Perhaps, could you add just a few words about what's happening below? It took me a bit of time to understand how the reverse line and regexp search work togeter.